code/game/mecha/combat/gorilla.dm

```diff
@@ -0,0 +1,198 @@
+	var/smoke_cooldown = 100
+	var/datum/effect/effect/system/smoke_spread/smoke_system = new
+	wreckage = /obj/effect/decal/mecha_wreckage/gorilla
+	add_req_access = 0
+	internal_damage_threshold = 25
+	force = 60
+	max_equip = 4
+	New()
+		..()
+		var/obj/item/mecha_parts/mecha_equipment/ME = new /obj/item/mecha_parts/mecha_equipment/tesla_energy_relay(src) // This thing basically cannot function without an external power supply.
+		ME.attach(src)
+		ME = new /obj/item/mecha_parts/mecha_equipment/weapon/ballistic/cannon(src)
+		ME.attach(src)
+		ME = new /obj/item/mecha_parts/mecha_equipment/weapon/ballistic/missile_rack/explosive(src)
+		ME.attach(src)
+		ME = new /obj/item/mecha_parts/mecha_equipment/weapon/ballistic/lmg(src)
+		ME.attach(src)
+		src.smoke_system.set_up(3, 0, src)
+		src.smoke_system.attach(src)
+		return
+
+/obj/mecha/combat/gorilla/mechstep(direction)
+	var/result = step(src,direction)
+	playsound(src,"mechstep",40,1)
+	return result
+
+/obj/mecha/combat/gorilla/mechturn(direction)
+	dir = direction
+	playsound(src,"mechstep",40,1)
+
+
+/obj/mecha/combat/gorilla/relaymove(mob/user,direction)
+	if(user != src.occupant)
+		user.loc = get_turf(src)
+		to_chat(user, "You climb out from [src]")
+		return 0
+	if(!can_move)
+		return 0
+	if(zoom)
+		if(world.time - last_message > 20)
+			src.occupant_message("Unable to move while in zoom mode.")
+			last_message = world.time
+		return 0
+	if(connected_port)
+		if(world.time - last_message > 20)
+			src.occupant_message("Unable to move while connected to the air system port")
+			last_message = world.time
+		return 0
+	if(state || !has_charge(step_energy_drain))
+		return 0
+	var/tmp_step_in = step_in
+	var/tmp_step_energy_drain = step_energy_drain
+	var/move_result = 0
+	if(internal_damage&MECHA_INT_CONTROL_LOST)
+		move_result = mechsteprand()
+	else if(src.dir!=direction)
+		move_result = mechturn(direction)
+	else
+		move_result	= mechstep(direction)
+	if(move_result)
+		if(istype(src.loc, /turf/space))
+			if(!src.check_for_support())
+				src.pr_inertial_movement.start(list(src,direction))
+		can_move = 0
+		spawn(tmp_step_in) can_move = 1
+		use_power(tmp_step_energy_drain)
+		return 1
+	return 0
+
+/obj/mecha/combat/gorilla/verb/smoke()
+	set category = "Exosuit Interface"
+	set name = "Smoke"
+	set src = usr.loc
+	set popup_menu = 0
+	if(usr!=src.occupant)
+		return
+	if(smoke_ready && smoke>0)
+		src.smoke_system.start()
+		smoke--
+		smoke_ready = 0
+		spawn(smoke_cooldown)
+			smoke_ready = 1
+	return
+
+/obj/mecha/combat/gorilla/verb/zoom()
+	set category = "Exosuit Interface"
+	set name = "Zoom"
+	set src = usr.loc
+	set popup_menu = 0
+	if(usr!=src.occupant)
+		return
+	if(src.occupant.client)
+		src.zoom = !src.zoom
+		src.log_message("Toggled zoom mode.")
+		src.occupant_message("<font color='[src.zoom?"blue":"red"]'>Zoom mode [zoom?"en":"dis"]abled.</font>")
+		if(zoom)
+			src.occupant.client.view = 12
+			src.occupant << sound('sound/mecha/imag_enh.ogg',volume=50)
```

Comment:
This should be `to_chat`, I believe.

Reply:
Wait, no, it's playing sound. Still, I think we have a helper for that.

Reply:
Will adjust to playsound in the next big commit.